CMakeLists.txt

```diff
@@ -84,6 +84,9 @@ if (EMSCRIPTEN)
             -fwasm-exceptions
             -pthread
             "SHELL:-s PTHREAD_POOL_SIZE=30"
+            "SHELL:-s FETCH=1"
+            "SHELL:-s ASSERTIONS=1"
+            "SHELL:-s USE_PTHREADS"
```

Comment:
I wonder what difference these make (Are they required for compilation? or correct behavior), and if these are something we should already have enabled independent of the HttpBlock?

Reply:
- `FETCH`: This is needed so that the emscripten fetch API can be used, this is also already enabled [in opencmw](https://github.com/fair-acc/opencmw-cpp/blob/a7a7c5c319b93ddfaf160893665a011d2d88bff8/CMakeLists.txt#L56)
- `ASSERTIONS: This is not strictly needed, but it is useful for debugging. This causes the printed stacktraces to be more meaningful (still not anywhere close to the debugging capabilities with a native build)
- `USE_PTHREADS`: This enables PTHREAD support, but apparently the `-pthread` flag above is already enough. I have removed this again.